src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -727,6 +736,29 @@ public static Map<String, String> parseTopicToTableMap(String input) {
     return topic2Table;
   }
 
+  /**
+   * Convert a Comma separated key value pairs into a Map
+   *
+   * @param input Provided in KC config
+   * @return Map
+   */
+  public static Map<String, String> parseCommaSeparatedKeyValuePairs(String input) {
+    Map<String, String> pairs = new HashMap<>();
+    for (String str : input.split(",")) {
+      String[] tt = str.split(":");
+
+      if (tt.length != 2 || tt[0].trim().isEmpty() || tt[1].trim().isEmpty()) {
+        LOGGER.error(
+            "Invalid {} config format: {}",
+            SnowflakeSinkConnectorConfig.SNOWPIPE_STREAMING_CLIENT_PROVIDER_OVERRIDE_MAP,
+            input);
+        throw SnowflakeErrors.ERROR_0030.getException();
```

Comment:
I think we should put the map as part of the exception instead of logging it? I remember we have customer complains about the needs of reading through the log to find the input